pkg/operator/rbaccontroller/rbaccontroller.go

```diff
@@ -0,0 +1,100 @@
+package rbaccontroller
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	listerrbacv1 "k8s.io/client-go/listers/rbac/v1"
+	"k8s.io/klog"
+)
+
+const (
+	// annotation used to indicate if unauthenticated user needs to be
+	// removed from the list of subjects for a cluster role binding
+	openShiftManagedRoleBindingAnnotation = "openshiftapiserver.rbac.authorization.openshift.io/managed"
+)
+
+// RBACController updates a cluster role binding after
+// removing unauthenticated user from the list of subjects
+type RBACController struct {
+	kubeClient               kubernetes.Interface
+	clusterRoleBindingLister listerrbacv1.ClusterRoleBindingLister
+	eventRecorder            events.Recorder
+}
+
+func NewRBACController(kubeClient kubernetes.Interface, informerFactory informers.SharedInformerFactory, recorder events.Recorder) factory.Controller {
+
+	c := &RBACController{
+		kubeClient:               kubeClient,
+		clusterRoleBindingLister: informerFactory.Rbac().V1().ClusterRoleBindings().Lister(),
+		eventRecorder:            recorder,
+	}
+
+	return factory.New().
+		WithInformers(informerFactory.Rbac().V1().ClusterRoleBindings().Informer()).
+		WithSync(c.sync).
+		ToController("RBACController", c.eventRecorder)
+}
+
+func (c *RBACController) sync(ctx context.Context, syncContext factory.SyncContext) error {
+	clusterRoleBindingNames := []string{"discovery", "system:basic-user", "system:discovery", "system:openshift:discovery", "cluster-status-binding"}
+
+	var errs []error
+
+	for _, clusterRoleBindingName := range clusterRoleBindingNames {
+		clusterRoleBinding, err := c.clusterRoleBindingLister.Get(clusterRoleBindingName)
+		if err != nil {
+			// Not every cluster role binding is present in every version of OpenShift.
+			// Downgrading the error to avoid log spamming
+			klog.V(4).Infof("fetching cluster role binding %s failed %v", clusterRoleBindingName, err)
+			continue
+		} else if clusterRoleBinding == nil {
+			continue
+		}
+
+		val, ok := clusterRoleBinding.Annotations[openShiftManagedRoleBindingAnnotation]
+		if !ok || val != "true" {
+			continue
+		}
+
+		updatedCopy, changed := withoutUnauthenticatedSubject(clusterRoleBinding)
+		if !changed {
+			continue
+		}
+
+		_, err = c.kubeClient.RbacV1().ClusterRoleBindings().Update(ctx, updatedCopy, metav1.UpdateOptions{})
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return errors.NewAggregate(errs)
+}
+
+// withoutUnauthenticatedSubject returns a new copy of given cluster role binding
+// without unauthenticated user in subjects
+func withoutUnauthenticatedSubject(clusterRoleBinding *rbacv1.ClusterRoleBinding) (*rbacv1.ClusterRoleBinding, bool) {
```

Comment:
this name was my suggestion. How about `withUnauthenticatedSubjectRemoved`?

Reply:
To me, `remove` seems like a more apt description of intent. 'with/without' seems better suited to the name of a method that provided a filtered view of an instance or field of that instance, since the instance/field are naturally the target of 'remove'.  But nevermind, this is clearly an exercise in bikeshedding.

Reply:
<programming-lang-bike-shed>removeXyz is imperative, withXyzRemove is functional</programming-lang-bike-shed>

Reply:
changed function name to `withUnauthenticatedSubjectRemoved`